Notebook updates: Some md -> Rmd, and colour -> color (#735)
AlexsLemonade/training-modules: scRNA-seq/02-filtering_scRNA.Rmd

```

Using the `coldata` data frame that we set up, we can now run the `tximeta()` to import our expression data while automatically finding and associating the transcript annotations that were used when we performed the quantification.
We'll also tell `tximeta()` to only read in the barcodes that `alevinQC` decided passed quality control.

The first time you run `tximeta()` you may get a message about storing downloaded transcriptome data in a cache directory so that it can retrieve the data more quickly the next time.
We recommend you use the cache, and accept the default location.

```{r read_data, live = TRUE}
# Read in alevin results with tximeta

> bladder_sce <- tximeta(coldata, type = "alevin", alevinArgs = list(filterBarcodes=TRUE))

Reviewer: Updating this means that we need to go through the notebook to find all the places where we talk about the number of cells and update those to the correct new values.

I can see right away that line 106 will need an update, but I suspect it is more extensive than that!

Minor style: we have been using spaces around `=` everywhere, so we will want that here, and may want to break this onto multiple lines

The first time you run `tximeta()` you may get a message about storing downloaded transcriptome data in a cache directory so that it can retrieve the data more quickly the next time.
We recommend you use the cache, and accept the default location.

```{r read_data, live = TRUE}
# Read in alevin results with tximeta
bladder_sce <- tximeta(coldata, type = "alevin")
```

A quick aside!

> When we ran `alevinQC` on this data in the last notebook, we saw that `salmon` had identified a "whitelist" of barcodes, i.e. barcodes that passed its quality control standards.

Reviewer: ```suggestion
When we ran `alevinQC` on this data in the last notebook, we saw that `salmon alevin` had identified a "whitelist" of barcodes that passed its quality control standards.
```

> Here, we are actually going to read in all barcodes, regardless of whether they are on the whitelist, so that we can explore different filtering thresholds ourselves (`salmon` can be very stringent when filtering out barcodes).

Reviewer: Breaking this up with reason first.
```suggestion
We could use this filtered list directly, but `salmon alevin` can be quite strict, and methods for filtering quite variable.
Instead, we will use the default behavior of `tximeta()` and read in all of the barcodes for which there is a non-zero UMI count (after barcode correction).
```

> But, if you wanted to read in the data considering only barcodes which passed `salmon`'s filter, you can supply the additional argument `alevinArgs = list(filterBarcodes=TRUE)` to the `tximeta()` function. 

Reviewer: ```suggestion
If you wanted instead to include only only barcodes that passed `salmon alevin`'s filter, you could supply the additional argument `alevinArgs = list(filterBarcodes=TRUE)` to the `tximeta()` function. 
```

> Even if you do choose to read in pre-filtered data approach, it's still important to explore the data as we're about to do here and potentially filter further based on your observations, in particular since mapping software's quality control measures (spoilers!) don't always filter based on mitochondrial gene content. 

Reviewer: ```suggestion
Even if you do choose to read in pre-filtered data, it's still important to explore the data as we're about to do here and potentially filter further based on your observations, in particular since mapping software's quality control measures (spoilers!) don't always filter based on mitochondrial gene content. 
```

  theme_classic()
```
Here, we want to keep cells with a low fraction of reads corresponding to mitochondrial genes and remove any cells with a high mitochondrial fraction.

> Again, it's important to take this step even if you have performed some initial filtering, since mapping software like `salmon`'s quality control does not consider mitochondrial read percentages.

Reviewer: I don't know we need to repeat this here, but if we do:
```suggestion
Again, it's important to take this step even if you are started with filtered data, since mapping software like `salmon alevin` and Cell Ranger do not usually consider mitochondrial read percentages when filtering.

```
